test/OrchardCore.Tests/Modules/OrchardCore.Resources/SubResourceIntegrityTests.cs

```diff
@@ -33,37 +33,42 @@ public async Task SavedSubResourceIntegritiesShouldMatchCurrentResources()
 
         async Task ValidateSubResourceIntegrityAsync(string resourceType)
         {
+            var expectations = new List<Tuple<string, string, string, string>>();
+
             foreach (var resource in resourceManifest.GetResources(resourceType))
             {
                 foreach (var resourceDefinition in resource.Value)
                 {
                     if (!string.IsNullOrEmpty(resourceDefinition.CdnIntegrity) && !string.IsNullOrEmpty(resourceDefinition.UrlCdnDebug))
                     {
-                        var resourceIntegrity = await GetSubResourceIntegrityAsync(httpClient, resourceDefinition.UrlCdnDebug);
-
-                        Assert.True(resourceIntegrity.Equals(resourceDefinition.CdnDebugIntegrity),
-                            $"The {resourceType} {resourceDefinition.UrlCdnDebug} has invalid SRI hash, please use '{resourceIntegrity}' instead.");
+                        var resourceIntegrityDebug = await GetSubResourceIntegrityAsync(httpClient, resourceDefinition.UrlCdnDebug);
+                        expectations.Add(new Tuple<string, string, string, string>(resourceType, resourceDefinition.UrlCdnDebug, resourceDefinition.CdnDebugIntegrity, resourceIntegrityDebug));
```

Comment:
Why not you the resource instead of a tuple?

Reply:
ah maybe pass the resourceDefinition as one object here. You need to generate the resourceIntegrity. And the resourceType doesn't need to be passed as it is a function param.